browser/src/control/Control.JSDialogBuilder.js

```diff
@@ -1517,7 +1517,7 @@ L.Control.JSDialogBuilder = L.Control.extend({
 	_editControl: function(parentContainer, data, builder, callback) {
 		var edit = L.DomUtil.create('input', 'ui-edit ' + builder.options.cssClass, parentContainer);
 		edit.value = data.text;
-		edit.id = data.id;
+		edit.id = data.id + '-input';
```

Comment:
This cannot be done. It will break updates for JSDialog. Main container/element of every widget has to have id, exactly like in `data.id`

Reply:
@eszkadev to bind label with with input element we need remove `-id` which are appended here in this commit to make as same as lable `htmlFor` https://github.com/CollaboraOnline/online/commit/8cd4bbd8b96952ef4ad35502b02b876bff9f92ad

And to handle that duplication of id we should add '-input' in main conatiner div and not for input fields

Thoughts ?

Reply:
Possibly we also can not use `arie-labelledby` everywhere because it conflicts with label and element connection.

Is it okay if we change ID of parent-container div and not append `-input` into to all id's of  `input` elements ?

Reply:
1. widget's main container always should have id = data.id
2. widget's main container's children can have any id you can imagine, but it has to be unique
3. you cannot modify parent inside widget's "function", `parentContainer` should only be used as a pointer where to put your widget. we can modify everything below in the tree (children) not above current widget (parent)

Reply:
@eszkadev understood your points, but my question why is it OK to change element id of

 `_radiobuttonControl` https://github.com/CollaboraOnline/online/blob/master/browser/src/control/Control.JSDialogBuilder.js#L1334, 
`_checkboxControl`,
https://github.com/CollaboraOnline/online/blob/master/browser/src/control/Control.JSDialogBuilder.js#L1382

AND NOT OK to change id of 

_editControl: https://github.com/CollaboraOnline/online/blob/master/browser/src/control/Control.JSDialogBuilder.js#L1599

> This cannot be done. It will break updates for JSDialog. Main container/element of every widget has to have id, exactly like in data.id

@eszkadev maybe i have added more confusion here but if widget will update for other control functions even if we changed id's of these elements then isn't it also ok for "edit Control".  It is also as same as other input tags, just difference is it's a editable input field 


If we should not change id here then there will be too much inconstancy for `input` elements, and it is very hard to establish relation between label and input tag